hirundo/dataset_optimization.py

```diff
@@ -370,7 +370,13 @@ def _check_run_by_id(run_id: str, retry=0) -> Generator[dict, None, None]:
                 last_event = json.loads(sse.data)
                 if not last_event:
                     continue
-                data = last_event["data"]
+                if "data" in last_event:
+                    data = last_event["data"]
+                else:
+                    if "detail" in last_event:
```

Comment:
What's this "detail"?

Reply:
This is an error "detail". It indicates what the error was

Reply:
This is an error "detail". It indicates what the error was